Add arrays, references and slices chapters.

atulkhare4096 opened the pull request:
Adds three related chapters on arrays, references and slices.

massand commented:
Few notes:
- [2-ownership](https://github.com/robertschaedler3/Fe2O3/blob/atulkhare/learn/2-ownership/README.md) already contains material about references and borrowing. Can we merge content in there somehow? Feel free to suggest replacements and align the [code snippets](https://github.com/robertschaedler3/Fe2O3/blob/atulkhare/learn/2-ownership/src/borrow.rs) with your material too. 
- Same with [slices](https://github.com/robertschaedler3/Fe2O3/blob/684e1b9a494cce4ab238c2b611c4e6479cccdfc5/learn/2-ownership/README.md#L162)

atulkhare4096 committed:
Modify ownership chapter

This tweaks the ownership chapter by shuffling a few things around.

1) References and slices are introduced before ownership.
2) Removed some now-redundant content from the ownership chapter.
